hackathon_site/hardware/views.py

```diff
@@ -31,9 +35,38 @@ def get(self, request, *args, **kwargs):
         return self.list(request, *args, **kwargs)
 
 
-class OrderListView(mixins.ListModelMixin, generics.GenericAPIView):
+class OrderViewSet(
+    mixins.ListModelMixin, mixins.CreateModelMixin, viewsets.GenericViewSet
+):
     queryset = Order.objects.all()
-    serializer_class = OrderSerializer
+    serializer_class = OrderListSerializer
+    serializer_action_classes = {
+        "list": OrderListSerializer,
+        "create": OrderCreateSerializer,
+    }
+
+    def get_serializer_class(self):
+        try:
+            return self.serializer_action_classes[self.action]
+        except (KeyError, AttributeError):
+            return super(OrderViewSet, self).get_serializer_class()
```

Comment:
Can just do `super().get_serializer_class()` now. We don't support old Python anyway.